core-primitives/substrate-sgx/externalities/src/lib.rs

```diff
@@ -147,9 +157,22 @@ where
 		self.state_diff.clear();
 	}
 
-	/// Execute the given closure while `self` is set as externalities.
-	///
-	/// Returns the result of the given closure.
+	fn clear_prefix(&mut self, key_prefix: &[u8], _maybe_limit: Option<u32>) -> u32 {
+		let to_remove = self
+			.state
+			.range::<[u8], _>((Bound::Included(key_prefix), Bound::Unbounded))
+			.map(|(k, _)| k)
+			.take_while(|k| k.starts_with(key_prefix))
+			.cloned()
+			.collect::<Vec<_>>();
+
+		let count = to_remove.len() as u32;
+		for key in to_remove {
+			self.remove(&key);
+		}
+		count
```

Comment:
Is this your own implementation, or is it copied from substrate?

Reply:
Okay, then we leave it as is. I was just wondering if a `drain_filter` would not be more efficient than what we see here. But that's just an implementation detail, and we trust substrate to know what they're doing 😉